hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/ReadableSpecialFileStore.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.admin;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+
+/**
+ * Provides read-only methods for interacting with the underlying data storage mechanisms for
+ * working with special files used in freeze transactions.
+ */
+public interface ReadableSpecialFileStore {
```

Comment:
FUnctionalInterface

Reply:
Not sure, if that would make sense here. In general, a store contains complex business logic and is not something you want to define as a lambda. I will leave it to the code owners of this service to decide.

Reply:
This interface is currently located in AdminService but it has been decided to move it to FileService. Once in FileService it will probably grow to have other methods.